tests/helpers/test_singleton.py

```diff
@@ -0,0 +1,40 @@
+"""Test singleton helper."""
+import pytest
+
+from homeassistant.helpers import singleton
+
+from tests.async_mock import Mock
+
+
+@pytest.fixture
+def mock_hass():
+    """Mock hass fixture."""
+    return Mock(data={})
+
+
+async def test_singleton_async(mock_hass):
+    """Test singleton with async function."""
+
+    @singleton.singleton("test_key")
+    async def something(hass):
+        return object()
+
+    result1 = await something(mock_hass)
+    result2 = await something(mock_hass)
+    assert result1 is result2
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6356951)